bio-qiime-pipeline/src/utils.py

```diff
@@ -0,0 +1,71 @@
+from typing import Optional, Tuple
+from pathlib import Path
+
+import csv
+
+from coretex import CustomSample, CustomDataset, Experiment
+from coretex.bioinformatics import ctx_qiime2
+
+
+def summarizeSample(sample: CustomSample, outputDir: Path) -> Path:
+    demuxPath = Path(sample.path) / "demux.qza"
+    visualizationPath = outputDir / "demux.qzv"
+
+    ctx_qiime2.demuxSummarize(str(demuxPath), str(visualizationPath))
+    return visualizationPath
+
+
+def determineTruncLen(sample: CustomSample, forward: bool) -> int:
+    sample.unzip()
+    summariesFileName = "forward-seven-number-summaries.tsv" if forward else "reverse-seven-number-summaries.tsv"
+    summariesTsv = list(sample.path.rglob(summariesFileName))[0]
+
+    truncLen: Optional[int] = None
+    with summariesTsv.open("r") as file:
+        summaries = list(csv.reader(file, delimiter = "\t"))
+
+    # summaries will allways have the median quality at row 5 of the csv
+    medianQualitiesStr = summaries[5]
+    medianQualitiesStr.pop(0)  # The first value will be "50%" and not a quality score
+    medianQualities = [float(x) for x in medianQualitiesStr]
+
+    highestScore = 0.0
```

Comment:
can't you use `max` on `medianQualities` to calculate this?

Reply:
The idea was for `highestScore` to track the highest quality score up to that point, so that scores are only compared to the ones that came before them, as `truncLen` works from left to right. Using `max` may work just fine, but seems less foolproof to me

Reply:
let's use builtin functions here instead of manually implementing the logic